### ISSUE TITLE
Multiple Inspector Windows/Tabs

### ISSUE THREAD

**Issue by SirPigeonz** [godotengine/godot-proposals#7459]
### Describe the project you are working on

Particle systems and other complex system of **nodes** and **resources** that require tweaks of multiple **properties** back and forth to achieve desired results.

### Describe the problem or limitation you are having in your project

To tweak or edit complex systems of nodes like for example particle system/particle scene file, user has to do a lot of scrolling to edit properties that are far apart and influence each other.

Example: Editing Material, Draw Passes Mesh and Particle Process Material properties to tweak or find desired look.

![image](https://github.com/godotengine/godot-proposals/assets/6152134/c343ba5d-d9c7-4163-8225-8c89a947e992)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If the user could open more than one Inspector Tab, it would minimize required scrolling to tweak properties.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Mockup how it could look.

![image](https://github.com/godotengine/godot-proposals/assets/6152134/a4d108cd-1635-408e-ac2b-2f3a039c5ef2)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It's an Editor usability problem, so a simple script would not help.

### Is there a reason why this should be core and not an add-on in the asset library?

It will often be a problem for anybody tweaking complex node and resource systems.
Especially artists that have to test multiple settings to find what they want or can achieve.

**Comment by Zireael07** [comment]
Problem: horizontal space can already be at a premium on some devices
However if this is easily toggleable/resettable to one pane, it would be a great addition for people on wide displays

**Comment by SirPigeonz** [comment]
> Problem: horizontal space can already be at a premium on some devices However if this is easily toggleable/resettable to one pane, it would be a great addition for people on wide displays

Yes, it would mostly be - use when needed, turn off when not - type of tool.

**Comment by zinnschlag** [comment]
Having a separate tab doesn't really fit into the design of the Godot editor UI, since it is composed of a fixed number of panels in fixed relative positions. Instead I would take the single inspector tab and split it horizontally into two inspectors.

So basically the original proposal, but with only a single "Inspector" tab on top and without breaking the established layout of the editor.

**Comment by Dark-Horizons** [comment]
Maybe, can be add Toggle Split mode like FileSystem Dock, instead of  new dock.
![image](https://github.com/godotengine/godot-proposals/assets/110924254/e025ffba-aef1-48e9-af60-67cb893f8e06)

**Comment by SirPigeonz** [comment]
> Maybe, can be add Toggle Split mode like FileSystem Dock, instead of new dock. This let us on two seperate inspector windows to work independently. ![image](https://user-images.githubusercontent.com/110924254/261835880-a2d4164e-b190-4173-b064-ddd91e85537a.png)

It would help, but problem is I sometimes need 3-4 (once even 5) inspectors next to each other to tweak stuff, especially particles. I would like to have an option to utilize horizontal screen space more.

**Comment by QbieShay** [comment]
I want to add that specifically for particle work this is a  big painpoint. Scrolling up and down is a lot of scrolling, and often you need to tweak material properties together with particle system's ones. this is very specific to particles work.

We've often floated around the idea of having a dedicated screen for it, but Godot's flexibility with it makes it really difficult to achieve unless we want to lock people into a specific workflow.

**Comment by SirPigeonz** [comment]
For editing multiple resources/nodes properties in Inspector, this is how Blender is doing it:
![image](https://github.com/godotengine/godot-proposals/assets/6152134/6a2b85c3-6e38-46c6-8655-a3be03e99ab9)

In the screen I have selected Cube, but I can still see and edit Suzanne properties in one of the "Inspectors"

They allow multiple Inspector editors (Property Editor) That has a "Pin" icon, so you can lock a specific window to always display your object props.

**Comment by SirPigeonz** [comment]
If somebody has better solutions open up. Maybe I should change the title of the proposal to something more vague tackling UX issue, not exact solution?

**Comment by robochase6000** [comment]
it's not clear from the proposal description, as it seems to focus on viewing the same object in both inspectors.  I think more important than that is the ability to view two objects side-by-side in two inspectors to compare differences.

the work around right now is to capture a screenshot of one, and open that side by side with the godot editor; it's pretty annoying to do this on a laptop as it requires resizing godot to make room for an image viewer program.

this is one thing I think Unity has right that should be considered: each inspector tab can be 'locked' to a specific object.  if an inspector tab isn't locked, it just shows the inspector for whatever object is currently selected. 

in addition to allowing side-by-side comparison, this kind of thing can open up some new workflows. i.e. if you have one node that you pretty much always want open, you lock that to one inspector and leave a second inspector for general browsing.

I could see this being not-so-simple the you start thinking about how have the 'open scene' tabs at the top that you can freely switch to, but I think this would be worth battling that complexity :D

**Comment by Spaz102** [comment]
This would be a great workflow improvement. The Unity editor supports something equivalent, and I used it a lot (Regardless, I'm never going back to Unity, lol).

The Godot editor has been getting a lot of tasty upgrades lately, so maybe there's already a solution in place (Or coming soon)? The side panels  **can** handle side-by-side docks (Exactly what we're asking for!), but I'm not seeing a way to open and lock a second Inspector.

And so, another use case for this request (My current pain point):
- I have a Label with all its layout/theme/etc properties tweaked to my needs
- I need to swap it for a RichTextLabel with all the same properties
- The current workflow is to switch the inspector back and forth between Controls _many times_, which is disorienting
- If I could open (and lock) a second Inspector, I could swiftly edit (and double-check) everything without _any_ switching

**Comment by Lex-DRL** [comment]
Coming from Unity with a big 4K TV as my main display, I must say that lack of this feature is one of the biggest pain points, even though I'm still just learning Godot. Can't imagine how tough it would be to **actually work** in Godot without an ability to open and lock/unlock 3-5 inspectors in 2 panels, with 5-15 Filesystem tabs in 3 other panels.

I think, the issue needs a general solution, not just a way to open, specifically, 2/3/4 inspectors.

I.e., Godot needs a standard dockable UI. Maybe, with a couple of tabs hard-coded to be in a specific panel (scene / script editor), but the rest of the UI needs to be fully cusomizeable.

**Comment by dustinlacewell** [comment]
This is easily _far and way_ the most painful aspect of using Godot as you are constantly grinding against it.

**Comment by Calinou** [comment]
@dustinlacewell Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by Ouren** [comment]
Please do this. I need to be able to edit the environment settings, lights and camera preview all at the same time.

**Comment by psychicteeth** [comment]
Yeah I would love this, I used it a lot in Unity and ran into a use case today.
I have a resource, that needs a default load-out of another type of resource. Godot's Resource class doesn't let you do that without some shenanigans that feel messy and unprofessional to me (something about making the resource a @tool, or saving each default subresource as a file and then preloading them). So I create a default resource with the data pre-filled out, already a data entry time sink. Next I want to duplicate this for each existing resource and copy over the data. I'd like to have one inspector with the new duplicate of the template and another inspector with the existing resource, and just drag the data over, or at least use it as a visual reference.

**Comment by ringh0f** [comment]
Yep, i really would like to have a second window of my choosing! This would improve my workflow tremendously.

**Comment by Hyperlooper** [comment]
Coming from Unity I ran into this right away. Would be great if we could arrive at an affordance for this in a way that doesn't break with Godots' UX philosophy. 

Another use-case is pinning the camera node so you can see the camera preview all of the time.
